app/recipes/tests/test_recipes_api.py

```diff
@@ -0,0 +1,259 @@
+            'name': 'Lemon cheesecake',
+            'description': 'Cheesecake (feat. lemon)',
+            'ingredients': [
+                {'name': 'Lemon'},
+                {'name': 'Cheese'}
+            ]
+        }
+
+        response = self.client.post(RECIPES_URL, payload, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(response.json(), {
+            'id': response.data['id'],
+            'name': 'Lemon cheesecake',
+            'description': 'Cheesecake (feat. lemon)',
+            'ingredients': [
+                {'name': 'Lemon'},
+                {'name': 'Cheese'}
+            ]
+        })
+        created_recipe = Recipe.objects.get(id=response.data['id'])
+        self.assertEqual(created_recipe.name, 'Lemon cheesecake')
+        self.assertEqual(created_recipe.description,
+                         'Cheesecake (feat. lemon)')
+        created_ingredients = Ingredient.objects.filter(recipe=created_recipe)
+        self.assertEqual(created_ingredients.count(), 2)
+        self.assertEqual(created_ingredients[0].name, 'Lemon')
+        self.assertEqual(created_ingredients[1].name, 'Cheese')
+
+    def test_create_recipe_without_description(self):
+        """POST /recipes/ should create recipe and return new resource
+            representation, including ID"""
+        payload = {
+            'name': 'Omelette',
+            'description': '',
+            'ingredients': [
+                {'name': 'Egg'},
+                {'name': 'Milk'},
+                {'name': 'Butter'},
+            ]
+        }
+
+        response = self.client.post(RECIPES_URL, payload, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(response.json(), {
+            'id': response.data['id'],
+            'name': 'Omelette',
+            'description': '',
+            'ingredients': [
+                {'name': 'Egg'},
+                {'name': 'Milk'},
+                {'name': 'Butter'},
+            ]
+        })
+        created_recipe = Recipe.objects.get(id=response.data['id'])
+        self.assertEqual(created_recipe.name, 'Omelette')
+        created_ingredients = Ingredient.objects.filter(recipe=created_recipe)
+        self.assertEqual(created_ingredients.count(), 3)
+        self.assertEqual(created_ingredients[0].name, 'Egg')
+        self.assertEqual(created_ingredients[1].name, 'Milk')
+        self.assertEqual(created_ingredients[2].name, 'Butter')
+
+    def test_create_recipe_invalid(self):
+        """POST /recipes/ with invalid data should return 400"""
+        payload = {
+            'type': 'Horse',
+            'legs': '4',
+            'speeds': ['TROT', 'CANTER', 'GALLOP']
+        }
+
+        response = self.client.post(RECIPES_URL, payload, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_read_recipes(self):
```

Comment:
A better name might be `test_list_recepies` and below instead of `test_read_recipe` change to `test_get_recipe`. Identifiers differing by only 1 character can be easily confused.